tests/experimental/test_mcmc.py

```diff
@@ -103,6 +114,31 @@ def test_MALA_regression_warmup(target: cuqi.density.Density):
     sampler_new = cuqi.experimental.mcmc.MALANew(target, scale=1)
     assert_true_if_warmup_is_equivalent(sampler_old, sampler_new)
 
+# # ============ CWMH ============
+
+@pytest.mark.parametrize("target", targets)
+def test_CWMH_regression_sample(target: cuqi.density.Density):
+    """Test the CWMH sampler regression."""
+    sampler_old = cuqi.sampler.CWMH(target, scale=np.ones(target.dim))
+    sampler_new = cuqi.experimental.mcmc.CWMHNew(target,
+                                                 scale=np.ones(target.dim))
+    assert_true_if_sampling_is_equivalent(sampler_old, sampler_new,
+                                          Ns=10,
+                                          old_idx=[0, -1],
+                                          new_idx=[1, -1])
+
+@pytest.mark.parametrize("target", targets)
+def test_CWMH_regression_warmup(target: cuqi.density.Density):
+    """Test the CWMH sampler regression."""
+    sampler_old = cuqi.sampler.CWMH(target, scale=np.ones(target.dim))
+    sampler_new = cuqi.experimental.mcmc.CWMHNew(target,
+                                                 scale=np.ones(target.dim))
+    Ns = 100 if target.dim < 50 else 20
+    assert_true_if_warmup_is_equivalent(sampler_old, sampler_new,
+                                        Ns=Ns,
+                                        strategy="MH_like",
+                                        old_idx=[0, -1],
+                                        new_idx=[1, None])
```

Comment:
Instead of old_idx and new_idx, which implicitly defines a range of values. I suggest for readabiltiy that we define an explicit range that we compare for the two samplers. E.g. one could compare [1..N] with [0..N-1] or similar. I think it would make the test much easier to understand.

Reply:
I agree, this makes it clearer. I actually thought about this, but it can make it tricky to set default sample size and default range since the range depends on the sample size. Also setting Ns and range separately, adds overhead on the developer to make them consistent, i.e. sat for the same N.

Reply:
Right. I think I got a bit confused. Perhaps this approach of using the indices for each set of samples makes sense. It may be helpful to add docstring with the parameters of this test.

Reply:
Great, I added documentation for these parameters